README.md

```diff
@@ -1,58 +1,148 @@
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  
+  <%= submit_tag "Submit Post" %>
+<% end %>
+```
 
-Bonus: Are we updating the entire post or only the fields the change?
+If you open the browser and go to the `edit` page it will now display the form, but you may have noticed a pretty big flaw, it doesn't load the record's data into the form, there are a few tasks that we'll need to do in order to implement this behavior, first let's have our `edit` action store the `post` record in an instance variable:
 
-in the event you only edit the title, does the post content get updated if we are always assigning all fields? dirty_attributes like content_changed? internalized in the sql for save/update (is that true?)
+```ruby
+def edit
+  @post = Post.find(params[:id])
+end
+```
+
+Now that the `edit` view template will have access to the `@post` object, we need to refactor the form so that it auto fills the form fields with the data from the `post` record, this is done below:
+
+```ERB
+<% # app/views/posts/edit.html.erb %>
+
+...
+<%= text_field_tag :title, @post.title %><br>
+...
+<%= text_area_tag :description, @post.description %><br>
+...
+```
+
+This will now populate the form, but if you tried to submit the form you may notice that it's redirecting to the show page and not changing the values. We need to first make a change to the form so that it knows what route the data should be passed through as well as what HTTP verb needs to be called, make the following change to the `form_tag` line:
+
+```ERB
+<%= form_tag post_path(@post), method: "put" do %>
+```
+
+By explicitly stating the route and method the form will now be routed to the `update` action. Before we try to implement the `update` action, let's first make sure the data is being routed properly. Enter the following code inside of the `update` method:
+
+```ruby
+def update
+  raise params.inspect
+end
+```
+
+If you open up the browser and navigate to an edit page, such as: `localhost:3000/post/5/edit` and change some elements in and form and submit it, it should take you to an error page that prints out the params from the form, such as the below image:
+
+![Raised Exception for Update Action](http://reif.io/lib/flatiron/update_raised_exception.png)
+
+As you can see, the parameters are being passed to the update action, with that in mind let's implement the functionality needed inside of the `update` action so that it will take the form data and update the specified record. Let's write some pseudo code for what the `update` action should do:
+
+* Query the database for the `Post` record that matched the `:id` passed to the route
+
+* Store the query in an instance variable
+
+* Update the values passed from the form
+
+* Save the changes in the database
+
+* Redirect the user to the `show` page so they can see the updated record
+
+
+We'll take advantage of the `update` method so that we're not manually assigning each attribute:
```

Comment:
clarify you mean the `update` method provided by ActiveRecord. Don't want people to confuse the `update` you're writing and the AR update